pydantic_ai_slim/pydantic_ai/ui/vercel_ai/_adapter.py

```diff
@@ -87,6 +88,57 @@ def messages(self) -> list[ModelMessage]:
         """Pydantic AI messages from the Vercel AI run input."""
         return self.load_messages(self.run_input.messages)
 
+    @cached_property
+    def deferred_tool_results(self) -> DeferredToolResults | None:
```

Comment:
Please make this a field on the superclass, like `messages`, `toolset`, etc.

Reply:
We should then also update `run_stream_native` to automatically pass through to `agent.run` in if it's set